### PR TITLE
Inform cargo of build script file / env var dependencies

### LINKED ISSUE #255: build script should notify cargo of their dependency on protocol files
Quoting @jplatte:

> You want to `println!("cargo:rerun-if-changed={}", protocol_file);` (see https://doc.rust-lang.org/cargo/reference/build-scripts.html#outputs-of-the-build-script)

So that the build script is re-run if the protocol file changes

### PR DESCRIPTION
**Pull request by jplatte**
fixes #255

### CODE REVIEW

File: wayland-client/build.rs [hunk 1]
```diff
@@ -10,5 +10,6 @@ fn main() {
     let out_dir_str = var("OUT_DIR").unwrap();
     let out_dir = Path::new(&out_dir_str);
 
+    println!("cargo:rerun-if-changed={}", protocol_file);
```
Comment (vberger) [inline wayland-client/build.rs]: You need to call `.display()` on the paths to get an object that can actually be fed to `println!()`.

**Comment by vberger** [comment]
Apart from the `Display` issue, lgtm. Thanks! :+1:

File: wayland-client/build.rs (hunk 1 above)
Comment (jplatte) [reply wayland-client/build.rs]: This is not the part of the code where `protocol_file` is a `&Path`, but you're right of course. I apparently didn't test locally :man_facepalming:

**Comment by vberger** [comment]
Thanks!

**Pull request merged**